### PR TITLE
Add "What's new" automation

### LINKED ISSUE #142: What's new in .NET Aspire docs - automation
---
[Associated WorkItem - 190725](https://dev.azure.com/msft-skilling/Content/_workitems/edit/190725)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add "What's new" automation.

Fixes #142

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/caching/caching-components.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/caching/caching-components.md) | [Tutorial: Implement caching with .NET Aspire components](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/caching-components?branch=pr-en-us-150) |
| [docs/caching/stackexchange-redis-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/caching/stackexchange-redis-component.md) | [.NET Aspire StackExchange Redis component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-component?branch=pr-en-us-150) |
| [docs/caching/stackexchange-redis-distributed-caching-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/caching/stackexchange-redis-distributed-caching-component.md) | [.NET Aspire StackExchange Redis distributed caching component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-distributed-caching-component?branch=pr-en-us-150) |
| [docs/caching/stackexchange-redis-output-caching-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/caching/stackexchange-redis-output-caching-component.md) | [docs/caching/stackexchange-redis-output-caching-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-output-caching-component?branch=pr-en-us-150) |
| [docs/database/azure-cosmos-db-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/azure-cosmos-db-component.md) | [.NET Aspire Azure Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-component?branch=pr-en-us-150) |
| [docs/database/azure-cosmos-db-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/azure-cosmos-db-entity-framework-component.md) | [.NET Aspire Microsoft Entity Framework Core Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-entity-framework-component?branch=pr-en-us-150) |
| [docs/database/postgresql-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/postgresql-component.md) | [.NET Aspire PostgreSQL component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-component?branch=pr-en-us-150) |
| [docs/database/postgresql-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/postgresql-entity-framework-component.md) | [.NET Aspire PostgreSQL Entity Framework Core component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-entity-framework-component?branch=pr-en-us-150) |
| [docs/database/sql-server-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/sql-server-component.md) | [docs/database/sql-server-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-component?branch=pr-en-us-150) |
| [docs/database/sql-server-components.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/sql-server-components.md) | [Tutorial: Connect an ASP.NET Core app to SQL Server using .NET Aspire and Entity Framework Core](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-components?branch=pr-en-us-150) |
| [docs/database/sql-server-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/database/sql-server-entity-framework-component.md) | [.NET Aspire SqlServer Entity Framework Core component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-entity-framework-component?branch=pr-en-us-150) |
| [docs/deployment/azure/aca-deployment-azd-in-depth.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/deployment/azure/aca-deployment-azd-in-depth.md) | [Deploy a .NET Aspire app to Azure Container Apps using the Azure Developer CLI (in-depth guide)](https://review.learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?branch=pr-en-us-150) |
| [docs/deployment/azure/application-insights.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/deployment/azure/application-insights.md) | [docs/deployment/azure/application-insights](https://review.learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/application-insights?branch=pr-en-us-150) |
| [docs/deployment/manifest-format.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/deployment/manifest-format.md) | [.NET Aspire manifest format for deployment tool builders](https://review.learn.microsoft.com/en-us/dotnet/aspire/deployment/manifest-format?branch=pr-en-us-150) |
| [docs/fundamentals/app-host-overview.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/app-host-overview.md) | [.NET Aspire orchestration overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/app-host-overview?branch=pr-en-us-150) |
| [docs/fundamentals/components-overview.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/components-overview.md) | [.NET Aspire components overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/components-overview?branch=pr-en-us-150) |
| [docs/fundamentals/dashboard.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/dashboard.md) | [docs/fundamentals/dashboard](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/dashboard?branch=pr-en-us-150) |
| [docs/fundamentals/health-checks.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/health-checks.md) | [Health checks in .NET Aspire](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/health-checks?branch=pr-en-us-150) |
| [docs/fundamentals/service-defaults.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/service-defaults.md) | [.NET Aspire service defaults](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/service-defaults?branch=pr-en-us-150) |
| [docs/fundamentals/setup-tooling.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/setup-tooling.md) | [.NET Aspire setup and tooling](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/setup-tooling?branch=pr-en-us-150) |
| [docs/fundamentals/telemetry.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/fundamentals/telemetry.md) | [docs/fundamentals/telemetry](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/telemetry?branch=pr-en-us-150) |
| [docs/get-started/add-aspire-existing-app.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/get-started/add-aspire-existing-app.md) | [Tutorial: Add .NET Aspire to an existing .NET 8 microservices app](https://review.learn.microsoft.com/en-us/dotnet/aspire/get-started/add-aspire-existing-app?branch=pr-en-us-150) |
| [docs/get-started/aspire-overview.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/get-started/aspire-overview.md) | [.NET Aspire overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview?branch=pr-en-us-150) |
| [docs/get-started/build-your-first-aspire-app.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/get-started/build-your-first-aspire-app.md) | [Quickstart: Build your first .NET Aspire app](https://review.learn.microsoft.com/en-us/dotnet/aspire/get-started/build-your-first-aspire-app?branch=pr-en-us-150) |
| [docs/messaging/azure-service-bus-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/messaging/azure-service-bus-component.md) | [docs/messaging/azure-service-bus-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-service-bus-component?branch=pr-en-us-150) |
| [docs/messaging/messaging-components.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/messaging/messaging-components.md) | [Tutorial: Use .NET Aspire messaging components in ASP.NET Core](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/messaging-components?branch=pr-en-us-150) |
| [docs/messaging/rabbitmq-client-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/messaging/rabbitmq-client-component.md) | [.NET Aspire RabbitMQ component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-client-component?branch=pr-en-us-150) |
| [docs/security/azure-security-key-vault-component.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/security/azure-security-key-vault-component.md) | [.NET Aspire Azure Key Vault component](https://review.learn.microsoft.com/en-us/dotnet/aspire/security/azure-security-key-vault-component?branch=pr-en-us-150) |
| [docs/service-discovery/overview.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/service-discovery/overview.md) | [TODO: Configuring polling interval and pending status refresh interval via `ServiceEndPointResolverOptions`](https://review.learn.microsoft.com/en-us/dotnet/aspire/service-discovery/overview?branch=pr-en-us-150) |
| [docs/storage/azure-storage-components.md](https://github.com/dotnet/docs-aspire/blob/0e16988b1ab68c27cb3a59a534746454f662e786/docs/storage/azure-storage-components.md) | [docs/storage/azure-storage-components](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-components?branch=pr-en-us-150) |

</details>

> **Note**
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs-aspire/pull/150/checks).

### CODE REVIEW

File: .whatsnew.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        "maximumNumberOfArticles": 3,
+        "tocParentNode": "Latest documentation updates",
+        "repoTocFolder": "docs/whats-new",
+        "indexParentNode": "Latest documentation updates",
+        "repoIndexFolder": "docs/whats-new"
+    },
+    "areas": [
+        {
+            "names": [ "fundamentals" ],
+            "heading": ".NET Aspire fundamentals"
```
Comment (BillWagner) [inline .whatsnew.json:19]: You might not want ".NET Aspire" in every heading. Every heading is for .NET Aspire, so it seems redundant.

File: docs/breadcrumb/toc.yml [hunk 2]
```diff
@@ -10,6 +10,9 @@ items:
           - name: .NET Aspire (Preview)
             tocHref: /dotnet/aspire/
             topicHref: /dotnet/aspire/index
+          - name: What's new in .NET Aspire
```
Comment (BillWagner) [inline docs/breadcrumb/toc.yml:13]: I might add "docs" here:

```suggestion
          - name: What's new in .NET Aspire Docs
```

File: .whatsnew.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        "relativeLinkPrefix": "../"
+    },
+    "navigationOptions": {
+        "maximumNumberOfArticles": 3,
+        "tocParentNode": "Latest documentation updates",
+        "repoTocFolder": "docs/whats-new",
+        "indexParentNode": "Latest documentation updates",
+        "repoIndexFolder": "docs/whats-new"
+    },
+    "areas": [
```
Comment (BillWagner) [inline .whatsnew.json:16]: The "areas" will be displayed in the final article in the order they are listed here. You may want to move some up or down. (I made a few suggestions below.

File: .whatsnew.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            "heading": ".NET Aspire security"
+        },
+        {
+            "names": [ "deployment" ],
+            "heading": ".NET Aspire deployment"
+        },
+        {
+            "names": [ "getting-started" ],
+            "heading": ".NET Aspire getting started"
+        },
```
Comment (BillWagner) [inline .whatsnew.json:48]: Maybe move this one up to the top?

**Review by BillWagner — APPROVED** [review]
This looks good to me. I had a suggestion or two, but otherwise, ship it.

**Comment by IEvangelist** [comment]
Related to #149

**Pull request merged**